core/trino-main/src/main/java/io/trino/operator/scalar/DataSizeFunctions.java

```diff
@@ -133,4 +172,140 @@ public static Unit parse(String unitString, String dataSize)
+                case "EB":
+                    return EXABYTE;
+                case "ZB":
+                    return ZETTABYTE;
+                case "YB":
+                    return YOTTABYTE;
+                default:
+                    throw invalidDataSize(dataSize);
+            }
+        }
+    }
+
+    @Description("Formats data size as bytes using a unit symbol")
+    @ScalarFunction(value = "format_data_size")
+    @SqlType(StandardTypes.VARCHAR)
+    public Slice formatDataSize(@SqlType(StandardTypes.BIGINT) long value)
+    {
+        return utf8Slice(formatLong((double) value, 1024));
+    }
+
+    @Description("Formats data size as bytes using a unit symbol")
+    @ScalarFunction(value = "format_data_size")
+    @SqlType(StandardTypes.VARCHAR)
+    public Slice formatDataSize(@SqlType(StandardTypes.DOUBLE) double value)
+    {
+        return utf8Slice(formatLong(value, 1024));
+    }
+
+    @Description("Formats data size as bytes using a unit symbol and a decimal base")
+    @ScalarFunction(value = "format_data_size_decimal")
+    @SqlType(StandardTypes.VARCHAR)
+    public Slice formatDataSizeDecimal(@SqlType(StandardTypes.BIGINT) long value)
+    {
+        return utf8Slice(formatLong((double) value, 1000));
+    }
+
+    @Description("Formats data size as bytes using a unit symbol and a decimal base")
+    @ScalarFunction(value = "format_data_size_decimal")
+    @SqlType(StandardTypes.VARCHAR)
+    public Slice formatDataSizeDecimal(@SqlType(StandardTypes.DOUBLE) double value)
+    {
+        return utf8Slice(formatLong(value, 1000));
+    }
+
+    private String formatLong(double fractional, long base)
+    {
+        String unit = "B";
+        if (fractional >= base || fractional <= -base) {
+            fractional /= base;
+            unit = "kB";
+        }
+        if (fractional >= base || fractional <= -base) {
+            fractional /= base;
+            unit = "MB";
+        }
+        if (fractional >= base || fractional <= -base) {
+            fractional /= base;
+            unit = "GB";
+        }
+        if (fractional >= base || fractional <= -base) {
+            fractional /= base;
+            unit = "TB";
+        }
+        if (fractional >= base || fractional <= -base) {
+            fractional /= base;
+            unit = "PB";
+        }
+        if (fractional >= base || fractional <= -base) {
+            fractional /= base;
+            unit = "EB";
+        }
+        if (fractional >= base || fractional <= -base) {
+            fractional /= base;
+            unit = "ZB";
+        }
+        if (fractional >= base || fractional <= -base) {
+            fractional /= base;
+            unit = "YB";
+        }
+
+        return getFormat(fractional).format(fractional) + unit;
+    }
+
+    private DecimalFormat getFormat(double value)
+    {
+        if (value < 10) {
+            // show up to two decimals to get 3 significant digits
+            return format3Number;
```

Comment:
This should account for negative values (`values > -10 && value < 10`)

Reply:
Added `Math.abs` around `value` and updated tests.